vg/vg.go

```diff
@@ -85,6 +88,49 @@ type CanvasSizer interface {
 	Size() (x, y Length)
 }
 
+// CanvasWriterTo wraps behavior required for writing
+// a Canvas representation to a concrete image.
+type CanvasWriterTo interface {
+	CanvasSizer
+	io.WriterTo
+}
+
+var (
+	formatLock sync.RWMutex
+	formats    = make(map[string]func(w, h Length) CanvasWriterTo)
```

Comment:
I'm really not sure about this. Can you explain what is required for opengl and html5/d3?

I don't really want to add a parameter that is not required for the majority of cases. Functional options may be OK if we have options be func(vg.Canvas) and the package-specific options are defined within the relevant package and assert to the correct type to perform the operation.